io-sim-classes/src/Control/Monad/Class/MonadSTM.hs

```diff
@@ -5,6 +5,18 @@
 module Control.Monad.Class.MonadSTM
   ( MonadSTM (..)
   , MonadSTMTx (..)
+  , MonadLabelledSTMTx (..)
+  , MonadLabelledSTM
+  , newLabelledTVar
+  , newLabelledTVarIO
+  , newLabelledTMVar
+  , newLabelledTMVarIO
+  , newLabelledEmptyTMVar
+  , newLabelledEmptyTMVarIO
+  , newLabelledTQueue
+  , newLabelledTQueueIO
+  , newLabelledTBQueue
+  , newLabelledTBQueueIO
```

Comment:
I'm not desperately happy about the profusion of these things. Do we really need these wrappers?

Can't we just always use the pattern of "create TThing; label TThing"?

If we need to be able to label in IO too, because of things like `newTVarIO` then we can have `labelTVarIO` that works in IO rather than STM.

Reply:
ok, let's get rid of them too.